otter/convergence/service.py

```diff
@@ -162,7 +162,7 @@ def all_met(server, current_lb_nodes):
                              if node.matches(server)]))
 
 
-def update_cache(group, servers, lb_nodes, now):
+def update_cache(group, servers, lb_nodes, now, include_deleted=True):
```

Comment:
This function as a separate unit probably does not know the right default for `include_deleted`. I think it would be better to take it as a positional argument. Moreover, calling it explicitly with `True` or `False` when gathering and succeeding makes it clear when deleted servers are cached.